nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/main/java/org/apache/nifi/distributed/cache/server/map/MapCacheServer.java

```diff
@@ -251,10 +251,17 @@ protected void finalize() throws Throwable {
     }
 
     private byte[] readValue(final DataInputStream dis) throws IOException {
-        final int numBytes = dis.readInt();
+        final int numBytes = validateInt(dis.readInt(), getMaxReadSize(), "readValue():");
         final byte[] buffer = new byte[numBytes];
         dis.readFully(buffer);
         return buffer;
     }
 
+    private int validateInt(final int value, final int max, final String identifier) throws IOException {
+        if (value <= max) {
+            return value;
+        } else {
+            throw new IOException(new IllegalArgumentException(identifier + value));
+        }
+    }
```

Comment:
Could this method be promoted to the superclass and shared in both `MapCacheServer` and `SetCacheServer`?

Reply:
I was thinking the same thing :)